modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotStorage.scala

```diff
@@ -93,10 +108,19 @@ object GlobalSnapshotStorage {
 
         def head: F[Option[Signed[GlobalSnapshotArtifact]]] = headRef.get
 
-        def get(ordinal: SnapshotOrdinal): F[Option[Signed[GlobalSnapshotArtifact]]] = cache(ordinal).get.flatMap {
-          case Some(s) => s.some.pure[F]
-          case None    => globalSnapshotLocalFileSystemStorage.read(ordinal)
-        }
+        def get(ordinal: SnapshotOrdinal): F[Option[Signed[GlobalSnapshotArtifact]]] =
+          ordinalCache(ordinal).get.flatMap {
+            _.fold(none[Signed[GlobalSnapshotArtifact]].pure[F])(get(_))
+          }.flatMap {
```

Comment:
Isn't this part already done in `get(hash: Hash)`? 🤔 I think you don't need to flatmap it again, right?

Reply:
yes, you're right